### DISCUSSION TITLE
[Detector Support]: How can Detect FPS exceed the frame rate of the underlying camera feed?

### DISCUSSION THREAD

**Discussion by pbix** [blakeblackshear/frigate#9899] [Detector Support]
### Describe the problem you are having

I want to start by saying that this is an really great project and I have been using it for several years and learned a lot in the process.  This post is not about a problem really but rather for my continuing education.

I spend time studying the meaning of the information on the Frigate system page.  In my setup I observe on occasion that the detect FPS exceeds the available camera frame rate.  I have observed this situation  often when it is windy and there is continuous motion in the field of view and only on exterior cameras.

I am confused.  How can the detect FPS be higher than the underlying camera feed?  What is the meaning of this (0 skipped) indication?

In my setup the main camera feed is set at 5 FPS (used for recording) in the camera and the low res sub feed is set at 2FPS (used for detect).  On the system screen in Frigate the capture feed shows 3.1 FPS and the Detect shows 6.8 FPS as show in the screen shot below.  The Detect FPS is the one the varies, the Capture never changes.  Its confusing that the numbers in Frigate do not agree with what is programmed in the camera.  This is an Amcrest camera.

If anyone can explain these discrepancy it would help me and perhaps others.

### Version

0.13.1-34fb1c2

### Frigate config file

```yaml
N/A
```

### docker-compose file or Docker CLI command

```yaml
N/A
```

### Relevant log output

```shell
N/A
```

### Operating system

Proxmox

### Install method

Docker Compose

### Coral version

CPU (no coral)

### Any other information that may be helpful

![09444](https://github.com/blakeblackshear/frigate/assets/289729/c45710a1-2d72-43e7-bb55-5de43123e2db)

**Comment by NickM-27** [comment]
Frigate runs detection on regions, which are portions of the camera frame. Sometimes these detections can be done multiple times on a single camera frame. In this case detection fps really means detections per second.

**Comment by realthk** [comment]
Confusing indeed. On a windy day, some of my cameras pick of many moving tree shadows, and this can even go over 40 FPS (that was the first clue before finding this topic that it cannot be real, no way the detect stream of the camera could have so many frames per second)

**Reply by hawkeye217** [reply]
In the next version of Frigate, this is labeled as "detections per second" to avoid this point of confusion.
